### PR TITLE
MSP-085-LIVE-R2: publish final 0.6.38 eight-leaf promotion lock

### LINKED ISSUE #416: MSP-085-LIVE-R2: publish final 0.6.38 eight-leaf promotion lock
## Summary\nThe official 0.6.38 HA deployment produced validator-accepted LIVE_CAPTURE M8 and M8.5 artifacts on gateway source 1e9c1d7ce19200a6dff79e32bc8ed6cf3ba6657c. Eight of the 18 frozen candidates are PROMOTED with LOCKED_NOT_EXPOSED; ten remain explicitly withheld. The current platform README still states that live completion is withheld.\n\n## Scope\n- publish the exact PUBLIC_REDACTED M8 evidence/report and M8.5 result bytes;\n- add a public completion record with source, binary, validation, restart-persistence, and count bindings;\n- update the platform index;\n- add focused tests for exact artifact hashes, public redaction, counts, and the M9-planning-only boundary.\n\n## Boundaries\n- no private campaign, raw capture, SKI, SHIP ID, eBUS/eeBUS addresses, trust-store bytes, or cryptographic secret material;\n- no GraphQL, Portal, Home Assistant consumer exposure, command routing, M9 implementation, V2/legacy alias, or candidate_ref;\n- no Fronius/SunSpec/modbusreg files;\n- no documentation in a code repository.\n\n## Acceptance\n- published files are byte-identical to validator-accepted outputs;\n- M8 report is PASS;\n- M8.5 is VALID_PROMOTION_LOCK with 8 promoted / 10 withheld and READY_FOR_M9_PLANNING;\n- every promoted candidate is LOCKED_NOT_EXPOSED;\n- public artifacts contain no raw operational identities or secrets;\n- focused tests and repository CI pass;\n- fresh exact-HEAD review returns NO_BLOCKING_FINDINGS.\n\n## Rollback\nRemove the public completion record and restore the prior index statement without changing runtime behavior or private evidence.

### PR DESCRIPTION
**Pull request by d3vi1**
Closes #416\n\n## Summary\n- publishes the exact PUBLIC_REDACTED M8 evidence/report and M8.5 result from the final 0.6.38 VR940/eBUS campaign\n- records the gateway source/binary and immutable artifact hashes\n- closes the obsolete live-withheld index state with 8 promoted / 10 withheld candidates\n- keeps every promoted candidate LOCKED_NOT_EXPOSED and M9 implementation out of scope\n\n## Validation\n- python3 -m pytest -q tests/test_msp085_live_completion.py (2 passed)\n- python3 -m pytest -q tests/test_multi_runtime_live_coexistence_contract.py tests/test_captured_multi_leaf_promotion_contract.py tests/test_leaf_promotion_dossier_contract.py (324 passed)\n- private M8 verifier PASS\n- private M8.5 verify-private PASS\n- bound derive-public + verify-public PASS\n\n## Boundaries\nNo private campaign, raw operational identity, secret material, GraphQL/Portal/HA consumer exposure, command routing, M9 implementation, or Fronius/SunSpec file is included.

### CODE REVIEW

**Comment by d3vi1** [comment]
Fresh exact-HEAD review on `fc4a72381ade95366db2697f937e3b0a97f24715`: `NO_BLOCKING_FINDINGS`. All three prior P2 findings are resolved. Hosted CI is green; canonical M8 public verification returned `public-only-ok`; focused validation passed.

File: scripts/validate_multi_runtime_coexistence.py [hunk 1]
```diff
@@ -1666,6 +1669,27 @@ def _source_portal(raw: dict[str, Any]) -> dict[str, Any]:
...
+)
+
+
+def _source_debug(value: Any) -> Any:
+    if isinstance(value, dict):
+        projected = {}
+        for key, item in value.items():
+            if key in SOURCE_DEBUG_ADDRESS_KEYS:
+                if isinstance(item, bool) or not isinstance(item, (int, str)):
+                    fail("provenance.source_capture")
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `9410279bac859cb47a61e6c03744e913f4e7f8e5`: the three declared nullable aliases preserve JSON `null`; only concrete numeric/text identities are pseudonymized. The public redactor accepts null only for these aliases. Added projection and redaction regression tests; 220 focused tests pass.

**Comment by d3vi1** [comment]
Final fresh exact-HEAD review on `9410279bac859cb47a61e6c03744e913f4e7f8e5`: `NO_BLOCKING_FINDINGS`. Nullable aliases preserve null while concrete identities are pseudonymized. All three hosted checks pass; focused validation: 220 passed.

File: docs/platform/live/msp-085-0.6.38/m8-evidence.json [hunk 2]
```diff
@@ -0,0 +1 @@
...f9c04dc","runtime":{"artifact_digest":"sha256:c71afac7eeeccdbf55ede21a5d14a437eb69492b02e35d10853b5b93cba33310","artifact_id":"gateway:sha256:c71afac7eeeccdbf55ede21a5d14a437eb69492b02e35d10853b5b93cba33310","artifact_size_bytes":35694375,"build_manifest":{"build_mode":"REPRODUCIBLE_BUILD","flags":["-trimpath","CGO_ENABLED=0"],"go_version":"go1.26.2","target":"linux/arm64"},"build_manifest_hash":"sha256:b4ad786c20d270b2cf8c1867c227dd94572eb0846c4bb91a3d88b394ec6b73bf","repository":"github.com/Project-Helianthus/helianthus-ebusgateway","source_commit":"1e9c1d7ce19200a6dff79e32bc8ed6cf3ba6657c","source_parent_commit":"8bcba2107d10b149f984ac9546ea6427a9cda8a1"}},"run_id":"msp08-run-04","state":"EEBUS_CONNECTED_ROLLBACK","state_evidence":{"candidate_count":0,"candidate_graph_enabled":false,"conflict_count":0,"degraded":false,"eebus_runtime_enabled":true,"empty_success":false,"facts":[],"outcome":"GRAPH_EVIDENCE_DROPPED","raw_only_count":0,"restart_transition":null,"service_count":1,"withheld_count":0}}],"schema_version":1,"scope":{"claims":["EEBUS-G18"],"excluded_gates":["EEBUS-G17","EEBUS-G19"],"gate":"EEBUS-G18","live_vr940_claim":true,"public_version_policy":"V1_ONLY_NO_PUBLIC_V2"}}
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply docs/platform/live/msp-085-0.6.38/m8-evidence.json]: Fixed in `bc915da9f3a26e29a9227fe5b22cf05f610c58a7`: all public eBUS address fields now use the constant `redacted:opaque-address`, including device addresses and admission aliases; device pseudonym hashes no longer include address. Null admission aliases remain null. Regenerated M8/M8.5 and exact hashes; private verify/derive/verify PASS; 333 focused tests pass.

File: scripts/validate_multi_runtime_coexistence.py [hunk 3]
```diff
@@ -1567,8 +1572,8 @@ def _source_device_projection(item: dict[str, Any]) -> dict[str, Any] | None:
     if not device_id or not manufacturer:
         fail("provenance.source_capture")
     return {
-        "address": _source_redacted(f"ebus-address:{address}"),
-        "device_id": _source_redacted(f"ebus-device:{address}:{device_id}"),
+        "address": OPAQUE_ADDRESS,
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1630]: Fixed in `dbd21ed9b5bd133b3b53ac78e4ae168cb606d050`: projection now retains the raw source address only as an in-memory sort key, drops it before publication, and derives distinct stable `via_device` pseudonyms from the already-redacted device identity. Added reversed-input determinism and distinct-via regression coverage. The canonical alias validator also normalizes spelling variants and recalculated-hash CLI tests cover device address plus all three nullable admission aliases. Private M8/M8.5 verify/derive/verify PASS; full local CI PASS.

File: scripts/validate_multi_runtime_coexistence.py [hunk 4]
```diff
@@ -2973,10 +3006,45 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+    return False
+
+
+def _contains_enumerable_public_address(evidence: dict[str, Any]) -> bool:
+    for run in evidence["runs"]:
+        views = {view["view_id"]: view["payload"] for view in run["protected_views"]}
+        if _contains_invalid_public_address(
+            views["mcp.ebus.v1.responses"], frozenset({"address"})
+        ) or _contains_invalid_public_address(
+            views["debug.ebus"], SOURCE_DEBUG_ADDRESS_KEYS, nullable=True
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `821cb8485e8862d3c60a0baad4e9514ffd1399ad`: captured public evidence now requires every `ha.identity.via_device` to equal the deterministic public-safe relation derived from the corresponding already-redacted MCP `device_id`. A canonical CLI mutation replaces `via_device` with an arbitrary/address-derived-style redacted hash, recalculates all payload/input/evidence hashes, and must fail `redaction.public`. Synthetic positive fixtures now use the same relationship; 226 focused tests and full local CI pass.

**Comment by d3vi1** [comment]
Review disposition for exact HEAD `dbd21ed9b5bd133b3b53ac78e4ae168cb606d050`: the request to make public CI reproduce the owner-private M8.5 campaign was independently evaluated once under the by-design rule and classified **BY_DESIGN_NONBLOCKING**. Public CI cannot independently reproduce withheld inputs; a public PASS receipt would remain owner self-attestation and add no verifiable property. The canonical owner-local gate was executed with the complete private source bundle (`verify-private` PASS, `derive-public`, bound `verify-public` PASS), while public CI locks exact artifact bytes, validates structure/redaction, and the public result binds the private campaign bytes plus M8 evidence/report hashes. Private inputs and command record remain retained outside public repositories. No attestation or private-input workflow is added.

File: scripts/validate_multi_runtime_coexistence.py [hunk 5]
```diff
@@ -1567,8 +1572,8 @@ def _source_device_projection(item: dict[str, Any]) -> dict[str, Any] | None:
     if not device_id or not manufacturer:
         fail("provenance.source_capture")
     return {
-        "address": _source_redacted(f"ebus-address:{address}"),
-        "device_id": _source_redacted(f"ebus-device:{address}:{device_id}"),
+        "address": OPAQUE_ADDRESS,
+        "device_id": _source_redacted(f"ebus-device:{device_id}"),
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1631]: Fixed in `044352a326c48c83ff6da4cd3745b1a8b96b57d4`: repeated source `device_id` values now receive stable non-enumerable per-occurrence public identities after address ordering, preserving MCP and HA multiplicity across input reordering. Public verification also rejects duplicate MCP device IDs, HA unique IDs, or HA via-device IDs after full hash recomputation. Added source-order and canonical CLI mutation regressions; 282 focused tests and full local CI pass.

File: scripts/validate_multi_runtime_coexistence.py [hunk 6]
```diff
@@ -2753,7 +2783,9 @@ def _valid_hash_like(value: Any) -> bool:
 def _valid_redacted_identity(value: Any) -> bool:
     if isinstance(value, list):
         return all(_valid_redacted_identity(item) for item in value)
-    return isinstance(value, str) and bool(REDACTED_ID_RE.fullmatch(value))
+    return isinstance(value, str) and bool(
+        REDACTED_ID_RE.fullmatch(value) or value == OPAQUE_ADDRESS
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:2889]: Fixed in `044352a326c48c83ff6da4cd3745b1a8b96b57d4`: the public validator now counts every `redacted:opaque-address` occurrence and accepts it only under declared MCP/debug address-bearing fields. A canonical CLI mutation moves it into every `meta.auth_subject`, recomputes payload/input/evidence hashes, and must fail `redaction.public`. 282 focused tests and full local CI pass.

File: scripts/validate_multi_runtime_coexistence.py [hunk 7]
```diff
@@ -1666,6 +1671,30 @@ def _source_portal(raw: dict[str, Any]) -> dict[str, Any]:
...
+    if isinstance(value, dict):
+        projected = {}
+        for key, item in value.items():
+            if key in SOURCE_DEBUG_ADDRESS_KEYS:
+                if item is None:
+                    projected[key] = None
+                elif isinstance(item, bool) or not isinstance(item, (int, str)):
+                    fail("provenance.source_capture")
+                else:
+                    projected[key] = OPAQUE_ADDRESS
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `044352a326c48c83ff6da4cd3745b1a8b96b57d4`: debug admission aliases now accept only `null` or strict eBUS address representations in range 0..255 (integer, decimal string, or one-byte hex string) before applying the opaque public placeholder. Projector regressions cover `-1`, `256`, `999`, `0x100`, and arbitrary text; a source-manifest-bound runtime test proves malformed private input fails `provenance.source_capture`. 282 focused tests, full local CI, and the owner-private M8/M8.5 chain pass.

File: scripts/validate_multi_runtime_coexistence.py [hunk 8]
```diff
@@ -2973,10 +3031,97 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+            views["debug.ebus"], SOURCE_DEBUG_ADDRESS_KEYS
+        )
+        device_ids = [item.get("device_id") for item in mcp_devices]
+        ha_unique_ids = [item.get("unique_id") for item in ha_devices]
+        ha_via_devices = [item.get("via_device") for item in ha_devices]
+        if (
+            len(mcp_devices) != len(ha_devices)
+            or any(
+                len(set(values)) != len(values)
+                for values in (device_ids, ha_unique_ids, ha_via_devices)
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:3435]: Fixed in 4da45dc2f7aebc5609927e80913728bc89c4e08d. The source projector now assigns canonical ordinal-bound device pseudonyms, and public verification recomputes each device_id plus the exact HA unique_id/via_device derivatives. A fully rehashed ebus-address:0..3 substitution now fails redaction.public; the live M8/M8.5 artifacts were regenerated and the bound private/public chain passes.

File: scripts/validate_multi_runtime_coexistence.py [hunk 9]
```diff
@@ -1666,6 +1671,44 @@ def _source_portal(raw: dict[str, Any]) -> dict[str, Any]:
...
+        return int(value, 16) <= 255
+    if re.fullmatch(r"(?:0|[1-9][0-9]{0,2})", value):
+        return int(value, 10) <= 255
+    return False
+
+
+def _source_debug(value: Any, path: tuple[str, ...] = ()) -> Any:
+    if isinstance(value, dict):
+        projected = {}
+        for key, item in value.items():
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1749]: Fixed in 4b748144f02fbbcc307ee014e4e80543be348e0e. Projection now synthesizes every missing canonical admission alias as JSON null while retaining and redacting present valid values. A partial selected_source-only source test proves the complete three-key output.

File: scripts/validate_multi_runtime_coexistence.py [hunk 10]
```diff
@@ -2753,7 +2809,9 @@ def _valid_hash_like(value: Any) -> bool:
 def _valid_redacted_identity(value: Any) -> bool:
     if isinstance(value, list):
         return all(_valid_redacted_identity(item) for item in value)
-    return isinstance(value, str) and bool(REDACTED_ID_RE.fullmatch(value))
+    return isinstance(value, str) and bool(
+        REDACTED_ID_RE.fullmatch(value) or value == OPAQUE_ADDRESS
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:2889]: Fixed in 4b748144f02fbbcc307ee014e4e80543be348e0e. Exact opaque-address path counting now runs for both SYNTHETIC_OFFLINE_FIXTURE and CAPTURED_RUNTIME_EVIDENCE; synthetic redacted address identities remain allowed only on canonical device address paths, while an opaque auth_subject in a fully rehashed synthetic artifact fails redaction.public.

File: scripts/validate_multi_runtime_coexistence.py [hunk 11]
```diff
@@ -2973,10 +3026,78 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+            len(mcp_devices) != len(ha_devices)
+            or device_ids != expected_device_ids
+            or any(
+                len(set(values)) != len(values)
+                for values in (device_ids, ha_unique_ids, ha_via_devices)
+            )
+            or any(
+                ha_device.get("unique_id")
+                != _source_redacted("ha:" + mcp_device["device_id"])
+                for mcp_device, ha_device in zip(mcp_devices, ha_devices, strict=True)
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:3440]: Fixed in 776029f4c9be24583ff45ede46e791b049fdfdfd. The projector derives the public model alias from the canonical ordinal-bound device_id; verify-public recomputes the expected model and requires exact equality in both MCP and HA identity views. The fully rehashed ebus-address:<n> model substitution now fails redaction.public. M8/M8.5 were regenerated and the complete source-bound chain plus local CI pass.

**Comment by d3vi1** [comment]
Fresh exact-HEAD review closure

- Reviewed HEAD: `776029f4c9be24583ff45ede46e791b049fdfdfd`
- Verdict: `NO_BLOCKING_FINDINGS`
- Focused review validation: 244 M8/live coexistence tests + 112 M8.5/promotion dossier tests
- Local canonical CI: PASS
- Hosted checks: 3/3 PASS
- All prior P0-P2 threads resolved; public/private LIVE_CAPTURE CI boundary remains independently validated `BY_DESIGN_NONBLOCKING`.
- Promotion remains `LOCKED_NOT_EXPOSED`; this merge does not enter M9.

File: scripts/validate_multi_runtime_coexistence.py [hunk 12]
```diff
@@ -2973,10 +3032,93 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+    return int(value == expected)
+
+
+def _contains_enumerable_public_address(evidence: dict[str, Any]) -> bool:
+    allowed_placeholder_count = 0
+    for run in evidence["runs"]:
+        views = {view["view_id"]: view["payload"] for view in run["protected_views"]}
+        mcp_devices = views["mcp.ebus.v1.responses"]["data"]["responses"][0][
+            "result"
+        ]["devices"]
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `dc2a62f3b8c119d21f3b03b18602f99a9e882613`. The validator now checks the MCP response shape and leading operation before dereference, returning the structured `redaction.public` category for reordered/malformed responses. Added a no-traceback regression. Focused M8/M8.5: 291 passed; full local CI passed.

[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `dc2a62f3b8c119d21f3b03b18602f99a9e882613`. Public coexistence validation now recursively rejects address/address-alias keys outside the exact canonical MCP device and debug-admission containers, including valid-looking redacted hashes. Added a regression that re-hashes the mutated evidence and proves `redaction.public`.

File: scripts/validate_multi_runtime_coexistence.py [hunk 13]
```diff
@@ -1666,6 +1671,47 @@ def _source_portal(raw: dict[str, Any]) -> dict[str, Any]:
...
+                    projected[key] = None
+                elif not _source_debug_address_valid(item):
+                    fail("provenance.source_capture")
+                else:
+                    projected[key] = OPAQUE_ADDRESS
+            else:
+                projected[key] = _source_debug(item, path + (key,))
+        if path == ("status", "admission"):
+            for key in SOURCE_DEBUG_ADDRESS_KEYS:
+                projected.setdefault(key, None)
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1769]: Fixed in `6a7b6c28612c3bfeb4d4492c2b884035fd40ac3d`. Source projection now synthesizes the canonical `status.admission` container with all three nullable aliases when either `admission` or the whole `status` container is absent, and rejects present containers with invalid types. Added RED/GREEN regressions for both missing-container cases. Focused M8/M8.5: 299 passed; full local CI passed; bound live artifacts remained byte-identical.

File: scripts/validate_multi_runtime_coexistence.py [hunk 14]
```diff
@@ -2973,10 +3050,200 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+        if not isinstance(admission, dict):
+            return True
+        normalized_debug_keys = {
+            _compact_key(key): key for key in SOURCE_DEBUG_ADDRESS_KEYS
+        }
+        for key, item in admission.items():
+            canonical_key = normalized_debug_keys.get(_compact_key(key))
+            if canonical_key is None:
+                continue
+            if key != canonical_key or item not in (None, OPAQUE_ADDRESS):
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:3398]: Fixed in `5bce18b9fe98fb8b555afa1aae238e547190ca90`. Captured-runtime public evidence now requires all three canonical nullable admission aliases; existing operational admission fields remain allowed. Added regressions for one missing alias, empty admission, and absent admission. Local official CI and private bound verification pass.

File: scripts/validate_multi_runtime_coexistence.py [hunk 15]
```diff
@@ -2973,10 +3050,200 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+        ]
+        expected_models = [
+            _source_redacted("ebus-model:" + device_id)
+            for device_id in expected_device_ids
+        ]
+        if (
+            len(mcp_devices) != len(ha_devices)
+            or device_ids != expected_device_ids
+            or [item.get("model") for item in mcp_devices] != expected_models
+            or [item.get("model") for item in ha_devices] != expected_models
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:3432]: Fixed in `5bce18b9fe98fb8b555afa1aae238e547190ca90`. Public validation now rejects any identity value equal to a deterministic hash from the complete 0..255 eBUS address domain, including `auth_subject`, while retaining only the canonical opaque address placeholder at allowed paths. Added a re-hashed auth-subject bypass regression.

File: scripts/validate_multi_runtime_coexistence.py [hunk 16]
```diff
@@ -2973,10 +3048,299 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+                "address",
+                "device_id",
+                "discovery_source",
+                "manufacturer",
+                "model",
+                "verification_state",
+            }
+            if any(
+                set(device) != expected_device_keys
+                or device.get("discovery_source") != "active_confirmed"
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed in `309e928ee13a88903eb10be6129c8da0aa23f445`. Captured public validation now accepts every canonical `SOURCE_DISCOVERY_STATES` value while still requiring `verification_state == identity_confirmed`. Added positive regressions for both `static_seed` and `passive_observed`. Official local CI is green, including 317 live M8/M8.5 tests and 304 executable-contract tests.

File: scripts/validate_multi_runtime_coexistence.py [hunk 17]
```diff
@@ -1666,6 +1671,63 @@ def _source_portal(raw: dict[str, Any]) -> dict[str, Any]:
...
+    if re.fullmatch(r"(?:0|[1-9][0-9]{0,2})", value):
+        return int(value, 10) <= 255
+    return False
+
+
+def _source_debug(value: Any, path: tuple[str, ...] = ()) -> Any:
+    if isinstance(value, dict):
+        projected = {}
+        for key, item in value.items():
+            if key in SOURCE_DEBUG_ADDRESS_KEYS:
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1750]: Fixed in `435f332cc22f72ee3bdebc7f27f55904db00816e`. `_source_debug` now rejects uncontracted address-like keys before projection, and the public verifier recognizes `addr`/`addrs` plus hash/digest suffixes. RED reproduced both the live projection and forged-public-artifact paths for `src_addr`, `dstAddr`, `peer_addrs`, and `source_addr_hash`; focused tests now pass, full local CI passes, and the bound private chain remains byte-identical.

[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py:1750]: Correction: the exact pushed commit is `435f332bd31dd997622c68dae0c62d21cb3d3730` (the abbreviated SHA remains `435f332`).

File: scripts/validate_multi_runtime_coexistence.py [hunk 18]
```diff
@@ -2973,10 +3072,335 @@ def _contains_public_secret(value: Any, key: str | None = None) -> bool:
...
+    if isinstance(value, list):
+        return sum(_count_exact_value(item, expected) for item in value)
+    return int(value == expected)
+
+
+def _is_public_address_key(key: Any) -> bool:
+    if not isinstance(key, str):
+        return False
+    compact = _compact_key(key)
+    return compact.endswith(
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply scripts/validate_multi_runtime_coexistence.py]: Fixed on current exact HEAD `c659bb2e736b25ed9af3db4ea951cab132f166d6`. Address keys are tokenized across snake_case and camelCase, so `src_addr_value`, `dstAddrRaw`, and `peer_addrs_list` are rejected in both source projection and forged public artifacts. The exact root `address_count` counter remains allowed only as a nonnegative integer. Full local CI and all three hosted checks pass; fresh exact-HEAD review returned `NO_BLOCKING_FINDINGS`.

**Comment by d3vi1** [comment]
Fresh exact-HEAD closure for `c659bb2e736b25ed9af3db4ea951cab132f166d6`: **NO_BLOCKING_FINDINGS**.

- All prior P0-P2 findings resolved; unresolved review threads: 0.
- Local canonical CI: PASS (`385/304/264/67/45/34`, Modbus `40+8`, opaque-runtime `230`).
- Hosted checks: 3/3 PASS.
- Bound owner-private replay: `verify-private` PASS; bound `verify-public` PASS.
- Public bytes unchanged: M8 evidence `9959206059a091a3805ab87e6dab6db753672c4fe03b453496260be2fb02b7c5`, M8 report `f211ab24f3718af21d6f414385f1554dacf654b610fb4f5b7695d57b0743424d`, M8.5 result `a0b41745f675e234f11729b2730bd640f04b673d4b8c97e3d0b56cc74ddb5b1e`.
- Promotion remains `LOCKED_NOT_EXPOSED`: 8 promoted, 10 withheld; M9 is not entered.
- Reviewer P3 for extra isolated numeric/counter regressions is triaged as backlog/nonblocking; direct probes and existing full-path tests pass.
